src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9773,11 +9773,27 @@ bool shouldTreatAsModifier()
                 using var _ = this.GetDisposableResetPoint(resetOnDispose: true);
 
                 Debug.Assert(this.CurrentToken.Kind == SyntaxKind.IdentifierToken);
-                this.EatToken();
 
-                return IsDeclarationModifier(this.CurrentToken.Kind) ||
-                    IsAdditionalLocalFunctionModifier(this.CurrentToken.Kind) ||
-                    (ScanType() != ScanTypeFlags.NotType && this.CurrentToken.Kind == SyntaxKind.IdentifierToken);
+                do
+                {
+                    this.EatToken();
+
+                    using var _2 = this.GetDisposableResetPoint(resetOnDispose: true);
+
+                    if (IsDeclarationModifier(this.CurrentToken.Kind) ||
+                        IsAdditionalLocalFunctionModifier(this.CurrentToken.Kind) ||
```

Comment:
>IsAdditionalLocalFunctionModifier(this.CurrentToken.Kind) [](http://example.com/codeflow?start=24&length=57)

Is the root of the problem the fact that ```IsAdditionalLocalFunctionModifier``` is supposed to be called with contextual kind and this is the only place where it isn't?

Reply:
As answered below, this would break some valid scenarios (`async async;`)